### PR TITLE
Contributing to Open Source document

### PR DESCRIPTION
**Pull request by lannonbr**
I am planning on a lightning talk to discuss how to contribute to the OSS community and I wrote this up to have as a resource for the talk and to look back at.

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Contributing to Open Source
+
+Contributing to Open Source allows people to be able to give back and help
+contribute to projects that they use frequenty. Whether adding a feature
```
Comment (emberian) [inline contributing.md]: s/contribute to/help out on/

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+to a project, fixing a bug, or adding documentation, it allows communities
```
Comment (emberian) [inline contributing.md]: Non-sequitur? This sentence is weird.

File: contributing.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+to form out of these projects.
+
+## Where to find projects
+
+Many widely known projects such as Firefox and VLC are open sourced and available
```
Comment (emberian) [inline contributing.md]: s/available to be contributed by members of the community/welcome contributions from newcomers./

File: contributing.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+Contributing to Open Source allows people to be able to give back and help
+contribute to projects that they use frequenty. Whether adding a feature
+to a project, fixing a bug, or adding documentation, it allows communities
+to form out of these projects.
+
+## Where to find projects
+
+Many widely known projects such as Firefox and VLC are open sourced and available 
+to be contributed by members of the community. Sites such as Github or Bitbucket
```
Comment (emberian) [inline contributing.md]: s/Github/GitHub

File: contributing.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+If there is a project that one uses frequently, it's possible to contribute
+to such project, although it's recommended to view if the Repository has
+a contributing file or license so one knows what they are allowed to do.
+
+If the project allows outside contributions, one is ready to help out.
+
+## How to contribute
+
+There are multiple ways of improving Open Source Software. A few of the main
+ways include writing of fixing documentation, posting bugs or feature requests,
```
Comment (emberian) [inline contributing.md]: s/of/or/

File: contributing.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+and even fixing part of the code itself. On Github, there is a button in the
```
Comment (emberian) [inline contributing.md]: s/Github/GitHub/

File: contributing.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+top corner of every repository that says Fork. By default a normal user can't
+edit code in repositories other than their own or ones they have permission to
+edit. Forking a repository allows the creation of personal copies of the
+repository on your own Github account so you can make changes to the codebase.
```
Comment (emberian) [inline contributing.md]: :%s/Github/GitHub/g

File: contributing.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+or a feature that may be confusing to understand and could be restructured
+to be easier to use.
+
+Users are allowed to either post an issue, or view and help fix an issue in
+the repo. These issues are threads that allow discussions on ways to fix
+bugs in software or make a better understanding on how the project works.
+
+## Documentation
+
+A resource that allows users of a piece of software to understand how to use
```
Comment (emberian) [inline contributing.md]: This paragraph is pretty messy! You should take another look at it.

File: contributing.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+contributors are often able to contribute to this documentation and help out
+the maintainers of the project, and the process is explained in the next section.
+
+## Branching and Pull Requests
+
+A convention that some people use when working on projects is to have a
+single branch of the codebase that is stable and different branches for
+adding new features to the repository. By default, a repository has a
+master branch when created, while other branches need to be created
+seperately.
```
Comment (emberian) [inline contributing.md:68]: You should link to https://guides.github.com/introduction/flow/

Comment (emberian) [inline contributing.md:68]: Also you should specify somewhere that this document is very git-centric, and ignores non-distributed version control systems entirely.

**Comment by lannonbr** [comment]
I have added and fixed the document based on suggestions above. Feel free to review again.

**Pull request merged**